R/utils.R

```diff
@@ -291,6 +291,20 @@ check_sampler_transitions_treedepth <- function(post_warmup_sampler_diagnostics,
   }
 }
 
+check_bfmi <- function(post_warmup_sampler_diagnostics) {
+  if (!is.null(post_warmup_sampler_diagnostics)) {
+    energy <- posterior::extract_variable_matrix(post_warmup_sampler_diagnostics, "energy__")
+    ebfmi <- apply(energy, 2, function(x) {
+      (sum(diff(x)^2)/length(x))/var(x)
+    })
+    if (any(ebfmi < .3)) {
+      message(sum(ebfmi < .3), " of ", length(ebfmi) , " chains had estimated Bayesian fraction
+      of missing information(E-BFMI) less than 0.3, which may indicate poor exploration of the 
```

Comment:
```suggestion
      message(sum(ebfmi < .3), " of ", length(ebfmi) , " chains had estimated Bayesian fraction
      of missing information (E-BFMI) less than 0.3, which may indicate poor exploration of the 
```

Reply:
Also maybe "energy-based Bayesian fraction of missing information"? Or some other name that includes "energy". It's weird to me to call it E-BFMI but ignore the E when writing out the name (I know it's like that in other places too but I think it's confusing in those places too).

Reply:
Unless the "E" refers to "estimated", but I always thought it referred to "energy"

Reply:
I always assumed e was for estimated, but p44 of the conceptual intro to HMC paper suggests that it refers to "energy", so I agree with you here.  

TBH I don't understand the first thing about the name of this quantity.  I have no idea why this quantity is a "fraction of missing information", not do I know why we want the fraction of missing information to be large (seems counterintuitive).  Maybe BFMI isn't the fraction of information that is missing, but rather the fraction of missing information that is Bayesian (like due to posterior uncertainty rather than sampler misbehavior)?

Reply:
Yeah I think it's a suboptimal name and it's confusing precisely for the reasons you mentioned. In fact I think I'm going to post on the forum about this and open an issue in the docs repo. I don't think there's anything in the Stan manual about BFMI but there should be an explanation that's clearer than the one in the conceptual intro, which is a great paper, but doesn't clarify this at all.

Reply:
Just posted: https://discourse.mc-stan.org/t/issues-with-e-bfmi-missing-doc-confusing-name-etc/22553. Hope it's ok that I mentioned I was motivated by chatting with you about this stuff.